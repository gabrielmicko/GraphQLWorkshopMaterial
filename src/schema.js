/**
 * Import makeExecutableSchema, resolvers, typeDefs
 * Call & export makeExecutableSchema, pass the imported variables
 */
import { makeExecutableSchema } from 'graphql-tools';
import resolvers from './resolvers';
import typeDefs from './types';

export default makeExecutableSchema({
  resolvers,
  typeDefs
});
