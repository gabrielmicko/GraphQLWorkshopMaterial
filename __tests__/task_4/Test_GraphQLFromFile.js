import { graphql } from 'graphql';
import schema from '../../src/schema';

test('GraphQL server should return one speaker by ID.', () => {
  expect.assertions(1);
  const query = /* GraphQL */ `
    {
      speakers(id: "cfc5f8de-b034-4c61-b885-1db70d319e1c") {
        id,
        bio,
        facebook,
        github,
        linkedin,
        twitter,
        position,
        name,
        photo,
        talks {
          id,
          speaker_id,
          lang,
          name,
          theme,
          title
        }
      }
    }
  `;
  return graphql(schema, query).then(result => {
    expect(result).toMatchSnapshot();
  });
});

test('GraphQL server should return one speaker by name.', () => {
  expect.assertions(1);
  const query = /* GraphQL */ `
    {
      speakers(name: "Gabriel Mičko") {
        id,
        bio,
        facebook,
        github,
        linkedin,
        twitter,
        position,
        name,
        photo,
        talks {
          id,
          speaker_id,
          lang,
          name,
          theme,
          title
        }
      }
    }
  `;
  return graphql(schema, query).then(result => {
    expect(result).toMatchSnapshot();
  });
});
