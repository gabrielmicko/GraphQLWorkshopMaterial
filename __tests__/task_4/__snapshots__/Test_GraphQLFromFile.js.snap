// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GraphQL server should return one speaker by ID. 1`] = `
Object {
  "data": Object {
    "speakers": Array [
      Object {
        "bio": "He is enthusiastic front-end developer who loves the web and new technologies around it. He is tech speaker, contributor, volunteer at Mozilla. He believes in the mission to ensure the Internet is a global public resource, open and accessible to all.",
        "facebook": "https://facebook.com/gabriel.micko",
        "github": "",
        "id": "cfc5f8de-b034-4c61-b885-1db70d319e1c",
        "linkedin": "",
        "name": "Gabriel Mičko",
        "photo": "http://fsa.jsday.com.br/img/people/gabriel_micko.jpg",
        "position": "Mozilla",
        "talks": Array [
          Object {
            "id": "6a5a8b27-121c-4dc1-96dd-3df5f065158b",
            "lang": "en",
            "name": "Gabriel Mičko",
            "speaker_id": "cfc5f8de-b034-4c61-b885-1db70d319e1c",
            "theme": "The presentation covers the main pillars of WebRTC which is video - audio calling and file sharing. It includes a well prepared demo with interaction with the audience, top-notch news and help to get started.",
            "title": "WebRTC From Zero to Hero",
          },
          Object {
            "id": "47c39051-0fba-4f24-a06b-0f4af92e670c",
            "lang": "en",
            "name": "Gabriel Mičko",
            "speaker_id": "cfc5f8de-b034-4c61-b885-1db70d319e1c",
            "theme": "The presentation covers an introduction to the GraphQL query language, explaining its features and benefits. I will show how minimal investment you have to do to start up your GraphQL endpoint without having to rewrite your entire database infrastructure. The workshop includes live demonstration (live coding).",
            "title": "GraphQL beyond the limit",
          },
        ],
        "twitter": "https://twitter.com/gabriel_micko",
      },
    ],
  },
}
`;

exports[`GraphQL server should return one speaker by name. 1`] = `
Object {
  "data": Object {
    "speakers": Array [
      Object {
        "bio": "He is enthusiastic front-end developer who loves the web and new technologies around it. He is tech speaker, contributor, volunteer at Mozilla. He believes in the mission to ensure the Internet is a global public resource, open and accessible to all.",
        "facebook": "https://facebook.com/gabriel.micko",
        "github": "",
        "id": "cfc5f8de-b034-4c61-b885-1db70d319e1c",
        "linkedin": "",
        "name": "Gabriel Mičko",
        "photo": "http://fsa.jsday.com.br/img/people/gabriel_micko.jpg",
        "position": "Mozilla",
        "talks": Array [
          Object {
            "id": "6a5a8b27-121c-4dc1-96dd-3df5f065158b",
            "lang": "en",
            "name": "Gabriel Mičko",
            "speaker_id": "cfc5f8de-b034-4c61-b885-1db70d319e1c",
            "theme": "The presentation covers the main pillars of WebRTC which is video - audio calling and file sharing. It includes a well prepared demo with interaction with the audience, top-notch news and help to get started.",
            "title": "WebRTC From Zero to Hero",
          },
          Object {
            "id": "47c39051-0fba-4f24-a06b-0f4af92e670c",
            "lang": "en",
            "name": "Gabriel Mičko",
            "speaker_id": "cfc5f8de-b034-4c61-b885-1db70d319e1c",
            "theme": "The presentation covers an introduction to the GraphQL query language, explaining its features and benefits. I will show how minimal investment you have to do to start up your GraphQL endpoint without having to rewrite your entire database infrastructure. The workshop includes live demonstration (live coding).",
            "title": "GraphQL beyond the limit",
          },
        ],
        "twitter": "https://twitter.com/gabriel_micko",
      },
    ],
  },
}
`;
